retrievalmarket/impl/provider_environments.go

```diff
@@ -217,7 +217,8 @@ type providerStoreGetter struct {
 
 func (psg *providerStoreGetter) Get(otherPeer peer.ID, dealID retrievalmarket.DealID) (*multistore.Store, error) {
 	var deal retrievalmarket.ProviderDealState
-	err := psg.p.stateMachines.GetSync(context.TODO(), retrievalmarket.ProviderDealIdentifier{Receiver: otherPeer, DealID: dealID}, &deal)
+	provDealID := retrievalmarket.ProviderDealIdentifier{Receiver: otherPeer, DealID: dealID}
+	err := psg.p.stateMachines.Get(provDealID).Get(&deal)
```

Comment:
Why have we replaced `GetSync` with `Get` here ?

Reply:
Good question, it was a bug that took me a while to track down.
While the deal moves to the unsealing state, the state machine calls a method to unseal and blocks until it returns. So no other events can be processed until unsealing completes.
If there is a restart, the client creates a new graphsync request, and on the response side, the provider needs to attach a datastore to the graphsync response. It gets the datastore by calling `providerStoreGetter.Get` (the method above). However if we call stateMachines.GetSync it will be blocked until unsealing complete, and the graphsync response times out, so the deal fails. We only need to access the state machine to get the StoreID, which is set when the state machine is initiated, so there's no need to use stateMachines.GetSync, we can just call stateMachines.Get, which doesn't get blocked.